### ISSUE TITLE
Sync at round preparing?

### ISSUE THREAD

**Issue by Binerax** [CaveeJohnson/wardrobe#9]
Hi, absolutely love this wardrobe system. My only problem is when people request a change the server lags for a second. This could be quite annoying doing rounds. Is there any way to only have models synced during the preparing phase? Before a round starts? Thank you again!

**Comment by CaveeJohnson** [comment]
Unfortunately not possible without significant (but relatively simple) changes, since you would need to add this factor into the client recieve code. The recieve code already deals with NULL players due to PVS dormancy, so you'd be complicating it quite a bit and maybe introducing bugs around it that it existed to solve.

https://github.com/CaveeJohnson/wardrobe/blob/master/lua/wardrobe/wardrobe.lua#L465

You want to put the `wardrobe.getAddon` call (everything after the 'already requested' shortcut) into a closure, and then store that closure in a table representing a queue, then when you're ready to deal with them (gamemode changes to prepare phase?) fire a function which processes the queue (pop each closure, execute it, could be parellelized but gmod makes that non trivial).

This should be relatively simple if you have familiarity with Lua, but I've also not tested it, so keep in mind the issues i mentioned.

**Comment by CaveeJohnson** [comment]
This is probably the simplest way, but technically the lag spike is the MOUNT not the DOWNLOAD, so this would just be delaying the downloads until round prep, if its a large file it could take into an ongoing round, then cause the lag when the MOUNT occurs.

Doing it the other way would probably be even messier.

**Comment by CaveeJohnson** [comment]
TL;DR

Not possible, but if you know lua you can probably do it without too much effort by queuing part of the process and processing the queue on the round change.

**Comment by Binerax** [comment]
Understood, If I had any lua experience I might've taken a jab at it but I don't really know lua. Thank you for the quick replies!
